rxandroidble/src/main/java/com/polidea/rxandroidble/ClientComponent.java

```diff
@@ -119,28 +118,50 @@ static int provideDeviceSdk() {
         }
 
         @Provides
-        @Named(NamedSchedulers.GATT_CALLBACK)
+        @Named(NamedSchedulers.BLUETOOTH_INTERACTION)
         @ClientScope
-        static ExecutorService provideGattCallbackExecutorService() {
+        static ExecutorService provideBluetoothInteractionExecutorService() {
             return Executors.newSingleThreadExecutor();
         }
 
         @Provides
-        @Named(NamedSchedulers.GATT_CALLBACK)
+        @Named(NamedSchedulers.BLUETOOTH_CALLBACKS)
         @ClientScope
-        static Scheduler provideGattCallbackScheduler(@Named(NamedSchedulers.GATT_CALLBACK) ExecutorService executorService) {
-            return Schedulers.from(executorService);
+        static ExecutorService provideBluetoothCallbacksExecutorService() {
+            return Executors.newSingleThreadExecutor();
         }
 
         @Provides
-        LocationManager provideLocationManager() {
-            return (LocationManager) context.getSystemService(Context.LOCATION_SERVICE);
+        @Named(NamedSchedulers.BLUETOOTH_INTERACTION)
+        @ClientScope
+        static Scheduler provideBluetoothInteractionScheduler(@Named(NamedSchedulers.BLUETOOTH_INTERACTION) ExecutorService service) {
+            return Schedulers.from(service);
+        }
+
+        @Provides
+        @Named(NamedSchedulers.BLUETOOTH_CALLBACKS)
+        @ClientScope
+        static Scheduler provideBluetoothCallbacksScheduler(@Named(NamedSchedulers.BLUETOOTH_CALLBACKS) ExecutorService service) {
+            return Schedulers.from(service);
         }
 
         @Provides
-        @Named(NamedSchedulers.MAIN_THREAD)
-        static Scheduler provideMainThreadScheduler() {
-            return AndroidSchedulers.mainThread();
+        static ClientComponentFinalizer provideFinalizationCloseable(
+                @Named(NamedSchedulers.BLUETOOTH_INTERACTION) final ExecutorService interactionExecutorService,
+                @Named(NamedSchedulers.BLUETOOTH_CALLBACKS) final ExecutorService callbacksExecutorService
+        ) {
+            return new ClientComponentFinalizer() {
+                @Override
+                public void onFinalize() {
+                    interactionExecutorService.shutdown();
```

Comment:
From what I see the executors are scoped. And `.shutdown()` is executed in `RxBleClient.finalize()`. Or am I missing something?